pinecone/core/grpc/index_grpc.py

```diff
@@ -155,42 +160,160 @@ def wrapped():
     def _request_metadata(self) -> Dict[str, str]:
         return {REQUEST_ID: _generate_request_id()}
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
+
+
+class vector(object):
+    def __init__(self, id: str, values: list, metadata: dict):
+        self.id = id
+        self.values = values
+        self.metadata = metadata
+
+    def __repr__(self):
+        return str({'id': self.id, 'values': self.values, 'metadata': self.metadata})
+
+
+class FetchResponse(object):
+    def __init__(self, vectors: dict, namespace: str):
+        self.vectors = vectors
+        self.namespace = namespace
+
+    def __repr__(self):
+        return str({'vectors': self.vectors, 'namespace': self.namespace})
+
+
+def parse_fetch_response(response: dict):
+    vd = {}
+    vectors = response.get('vectors')
+    if not vectors:
+        return None
```

Comment:
This might be breaking compatibility with the openapi client - maybe there it returns an object with an empty vectors field?